src/v2/manifest/manifest_schema1.rs

```diff
@@ -49,4 +49,35 @@ impl ManifestSchema1Signed {
             .map(|l| l.blob_sum.clone())
             .collect()
     }
+
+    /// List all image labels stored in the history array of this manifest.
+    ///
+    /// The returned list has no ordering guarantees.
+    pub fn get_labels(&self) -> Option<(Vec<(String, String)>)> {
+        if let Ok(v1_compat_top) =
+            serde_json::from_str::<serde_json::Value>(&self.history[0].v1_compat)
+        {
+            if let Some(container_config) = v1_compat_top.get("container_config") {
+                if let Some(labels) = container_config.get("Labels") {
+                    if let Some(labels) = labels.as_object() {
+                        let mut v = Vec::new();
```

Comment:
In general, this is not a nice idiom. It is preferred to iterate and `filter_map+collect`, or at least to construct using `with_capacity` and `shrink_to_fit` at the end.